src/Server/grpc_protos/clickhouse_grpc.proto

```diff
@@ -41,6 +41,25 @@ message ExternalTable {
    map<string, string> settings = 5;
 }
 
+enum CompressionAlgorithm {
+   NO_COMPRESSION = 0;
+   DEFLATE = 1;
+   GZIP = 2;
+   STREAM_GZIP = 3;
+}
+
+enum CompressionLevel {
```

Comment:
As far as I remember compression level for LZ4 for example could be an arbitrary number, for example 7. It is useless, but maybe we have to support?